<?xml version="1.0"?>

<robot name="a1" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:arg name="DEBUG" default="false"/>

    <xacro:include filename="$(find a1_description)/xacro/const.xacro"/>
    <xacro:include filename="$(find a1_description)/xacro/materials.xacro"/>
    <xacro:include filename="$(find a1_description)/xacro/leg.xacro"/>
    <xacro:include filename="$(find a1_description)/xacro/stairs.xacro"/>
    <xacro:include filename="$(find a1_description)/xacro/gazebo.xacro"/>
    <!-- <xacro:include filename="$(find a1_gazebo)/launch/stairs.urdf.xacro"/> -->

    <xacro:include filename="$(find a1_description)/xacro/transmission.xacro"/>

    <!-- Initialize camera-->
    <xacro:include filename="$(find a1_description)/xacro/camera.xacro"/>

    <!-- <xacro:stairs stairs="15" xpos="0" ypos="0" zpos="0" /> -->

    <!-- Rollover Protection mode will add an additional stick on the top, use "true" or "false" to switch it. -->
    <xacro:property name="rolloverProtection" value="false"/>

    <!-- Debug mode will hung up the robot, use "true" or "false" to switch it. -->
    <xacro:if value="$(arg DEBUG)">
        <link name="world"/>
        <joint name="base_static_joint" type="fixed">
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <parent link="world"/>
            <child link="base"/>
        </joint>
    </xacro:if> 

    <link name="base">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="0.001 0.001 0.001"/>
            </geometry>
        </visual>
    </link>

    <joint name="floating_base" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="base"/>
        <child link="trunk"/>
    </joint>

    <link name="trunk">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://a1_description/meshes/trunk.dae" scale="1 1 1"/>
            </geometry>
            <material name="orange"/>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="${trunk_length} ${trunk_width} ${trunk_height}"/>
            </geometry>
        </collision>
        <inertial>
            <origin rpy="0 0 0" xyz="${trunk_com_x} ${trunk_com_y} ${trunk_com_z}"/>
            <mass value="${trunk_mass}"/>
            <inertia
                ixx="${trunk_ixx}" ixy="${trunk_ixy}" ixz="${trunk_ixz}"
                iyy="${trunk_iyy}" iyz="${trunk_iyz}"
                izz="${trunk_izz}"/>  
        </inertial>
    </link>

    <xacro:realsense2_camera parent="trunk" name="d435i_camera"/>

    <xacro:if value="${(rolloverProtection == 'true')}">
        <joint name="stick_joint" type="fixed">
            <parent link="trunk"/>
            <child link="stick_link"/>
            <origin rpy="0 0 0" xyz="${0.18} 0 ${stick_length/2.0+0.08}"/>
        </joint>

        <link name="stick_link">
            <visual>
                <geometry>
                    <cylinder length="${stick_length}" radius="${stick_radius}"/>
                </geometry>
                <origin rpy="0 0 0" xyz="0 0 0"/>
                <material name="white"/>
            </visual>
            <collision>
                <geometry>
                    <cylinder length="${stick_length}" radius="${stick_radius}"/>
                </geometry>
                <origin rpy="0 0 0" xyz="0 0 0"/>
            </collision>
            <inertial>
                <mass value="${stick_mass}"/>
                <inertia
                    ixx="${stick_mass / 2.0 * (stick_radius*stick_radius)}" ixy="0.0" ixz="0.0"
                    iyy="${stick_mass / 12.0 * (3*stick_radius*stick_radius + stick_length*stick_length)}" iyz="0.0"
                    izz="${stick_mass / 12.0 * (3*stick_radius*stick_radius + stick_length*stick_length)}"/>                   
            </inertial>
        </link>
    </xacro:if> 

    <joint name="imu_joint" type="fixed">
        <parent link="trunk"/>
        <child link="imu_link"/>
        <origin rpy="0 0 0" xyz="0 0 0"/>
    </joint>

    <link name="imu_link">
        <inertial>
            <mass value="0.001"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="0.001 0.001 0.001"/>
            </geometry>
            <material name="red"/>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size=".001 .001 .001"/>
            </geometry>
        </collision>
    </link>

    <!-- DON'T USE THIS, modified axes for locomotion policy -->
    <!-- <xacro:leg name="FR" mirror="-1" mirror_dae="False" front_hind="1" front_hind_dae="True">
        <origin rpy="0 0 0" xyz="${leg_offset_x} ${-leg_offset_y} 0"/>
    </xacro:leg>

    <xacro:leg name="FL" mirror="1" mirror_dae="True" front_hind="1" front_hind_dae="True">
        <origin rpy="0 0 0" xyz="${leg_offset_x} ${leg_offset_y} 0"/>
    </xacro:leg>

    <xacro:leg name="RR" mirror="-1" mirror_dae="False" front_hind="-1" front_hind_dae="False">
        <origin rpy="0 0 0" xyz="${-leg_offset_x} ${-leg_offset_y} 0"/>
    </xacro:leg>

    <xacro:leg name="RL" mirror="1" mirror_dae="True" front_hind="-1" front_hind_dae="False">
        <origin rpy="0 0 0" xyz="${-leg_offset_x} ${leg_offset_y} 0"/>
    </xacro:leg> -->
    <joint name="FR_hip_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0.1805 -0.047 0"/>
        <parent link="trunk"/>
        <child link="FR_hip"/>
        <axis xyz="1 0 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-0.8028514559173915" upper="0.8028514559173915" velocity="21"/>
    </joint>
    <link name="FR_hip">
        <visual>
        <origin rpy="3.141592653589793 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/hip.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.04" radius="0.046"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="-0.003875 -0.001622 4.2e-05"/>
        <mass value="0.595"/>
        <inertia ixx="0.000402747" ixy="8.709e-06" ixz="-2.97e-07" iyy="0.000691123" iyz="5.45e-07" izz="0.000487919"/>
        </inertial>
    </link>
    <joint name="FR_hip_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 -0.081 0"/>
        <parent link="FR_hip"/>
        <child link="FR_thigh_shoulder"/>
    </joint>
    <!-- this link is only for collision -->
    <link name="FR_thigh_shoulder">
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.032" radius="0.041"/>
        </geometry>
        </collision>
    </link>
    <joint name="FR_thigh_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 -0.0838 0"/>
        <parent link="FR_hip"/>
        <child link="FR_thigh"/>
        <axis xyz="0 -1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-4.1887902047863905" upper="1.0471975511965976"  velocity="21"/>
    </joint>
    <link name="FR_thigh">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/thigh_mirror.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.0245 0.034"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="-0.003574 0.019529 -0.030323"/>
        <mass value="0.888"/>
        <inertia ixx="0.005251806" ixy="2.168e-06" ixz="0.000346889" iyy="0.005000475" iyz="2.8174e-05" izz="0.0011102"/>
        </inertial>
    </link>
    <joint name="FR_calf_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="FR_thigh"/>
        <child link="FR_calf"/>
        <axis xyz="0 -1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="0.9162978572970231" upper="2.6965336943312392" velocity="21"/>
    </joint>
    <link name="FR_calf">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/calf.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.016 0.016"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0.007105 -0.000239 -0.096933"/>
        <mass value="0.151"/>
        <inertia ixx="0.002344758" ixy="0.0" ixz="-0.000141275" iyy="0.002360755" iyz="0.0" izz="3.1158e-05"/>
        </inertial>
    </link>
    <joint name="FR_foot_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="FR_calf"/>
        <child link="FR_foot"/>
    </joint>
    <link name="FR_foot">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.02"/>
        </geometry>
        </collision>
        <inertial>
        <mass value="0.06"/>
        <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
        </inertial>
    </link>
    <xacro:leg_transmission name="FR"/>

    <joint name="FL_hip_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0.1805 0.047 0"/>
        <parent link="trunk"/>
        <child link="FL_hip"/>
        <axis xyz="1 0 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-0.8028514559173915" upper="0.8028514559173915" velocity="21"/>
    </joint>
    <link name="FL_hip">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/hip.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.04" radius="0.046"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="-0.003875 0.001622 4.2e-05"/>
        <mass value="0.595"/>
        <inertia ixx="0.000402747" ixy="-8.709e-06" ixz="-2.97e-07" iyy="0.000691123" iyz="-5.45e-07" izz="0.000487919"/>
        </inertial>
    </link>
    <joint name="FL_hip_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 0.081 0"/>
        <parent link="FL_hip"/>
        <child link="FL_thigh_shoulder"/>
    </joint>
    <!-- this link is only for collision -->
    <link name="FL_thigh_shoulder">
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.032" radius="0.041"/>
        </geometry>
        </collision>
    </link>
    <joint name="FL_thigh_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 0.0838 0"/>
        <parent link="FL_hip"/>
        <child link="FL_thigh"/>
        <axis xyz="0 1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-1.0471975511965976" upper="4.1887902047863905" velocity="21"/>
    </joint>
    <link name="FL_thigh">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/thigh.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.0245 0.034"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="-0.003574 -0.019529 -0.030323"/>
        <mass value="0.888"/>
        <inertia ixx="0.005251806" ixy="-2.168e-06" ixz="0.000346889" iyy="0.005000475" iyz="-2.8174e-05" izz="0.0011102"/>
        </inertial>
    </link>
    <joint name="FL_calf_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="FL_thigh"/>
        <child link="FL_calf"/>
        <axis xyz="0 1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-2.6965336943312392" upper="-0.9162978572970231" velocity="21"/>
    </joint>
    <link name="FL_calf">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/calf.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.016 0.016"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0.007105 -0.000239 -0.096933"/>
        <mass value="0.151"/>
        <inertia ixx="0.002344758" ixy="0.0" ixz="-0.000141275" iyy="0.002360755" iyz="0.0" izz="3.1158e-05"/>
        </inertial>
    </link>
    <joint name="FL_foot_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="FL_calf"/>
        <child link="FL_foot"/>
    </joint>
    <link name="FL_foot">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.02"/>
        </geometry>
        </collision>
        <inertial>
        <mass value="0.06"/>
        <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
        </inertial>
    </link>
    <xacro:leg_transmission name="FL"/>


    <joint name="RR_hip_joint" type="revolute">
        <origin rpy="0 0 0" xyz="-0.1805 -0.047 0"/>
        <parent link="trunk"/>
        <child link="RR_hip"/>
        <axis xyz="-1 0 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-0.8028514559173915" upper="0.8028514559173915" velocity="21"/>
    </joint>
    <link name="RR_hip">
        <visual>
        <origin rpy="3.141592653589793 3.141592653589793 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/hip.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.04" radius="0.046"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0.003875 -0.001622 4.2e-05"/>
        <mass value="0.595"/>
        <inertia ixx="0.000402747" ixy="-8.709e-06" ixz="2.97e-07" iyy="0.000691123" iyz="5.45e-07" izz="0.000487919"/>
        </inertial>
    </link>
    <joint name="RR_hip_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 -0.081 0"/>
        <parent link="RR_hip"/>
        <child link="RR_thigh_shoulder"/>
    </joint>
    <!-- this link is only for collision -->
    <link name="RR_thigh_shoulder">
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.032" radius="0.041"/>
        </geometry>
        </collision>
    </link>
    <joint name="RR_thigh_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 -0.0838 0"/>
        <parent link="RR_hip"/>
        <child link="RR_thigh"/>
        <axis xyz="0 -1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-4.1887902047863905" upper="1.0471975511965976" velocity="21"/>
    </joint>
    <link name="RR_thigh">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/thigh_mirror.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.0245 0.034"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="-0.003574 0.019529 -0.030323"/>
        <mass value="0.888"/>
        <inertia ixx="0.005251806" ixy="2.168e-06" ixz="0.000346889" iyy="0.005000475" iyz="2.8174e-05" izz="0.0011102"/>
        </inertial>
    </link>
    <joint name="RR_calf_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="RR_thigh"/>
        <child link="RR_calf"/>
        <axis xyz="0 -1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="0.9162978572970231" upper="2.6965336943312392" velocity="21"/>
    </joint>
    <link name="RR_calf">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/calf.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.016 0.016"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0.007105 -0.000239 -0.096933"/>
        <mass value="0.151"/>
        <inertia ixx="0.002344758" ixy="0.0" ixz="-0.000141275" iyy="0.002360755" iyz="0.0" izz="3.1158e-05"/>
        </inertial>
    </link>
    <joint name="RR_foot_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="RR_calf"/>
        <child link="RR_foot"/>
    </joint>
    <link name="RR_foot">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.02"/>
        </geometry>
        </collision>
        <inertial>
        <mass value="0.06"/>
        <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
        </inertial>
    </link>
    <xacro:leg_transmission name="RR"/>


    <joint name="RL_hip_joint" type="revolute">
        <origin rpy="0 0 0" xyz="-0.1805 0.047 0"/>
        <parent link="trunk"/>
        <child link="RL_hip"/>
        <axis xyz="-1 0 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-0.8028514559173915" upper="0.8028514559173915" velocity="21"/>
    </joint>
    <link name="RL_hip">
        <visual>
        <origin rpy="0 3.141592653589793 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/hip.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.04" radius="0.046"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0.003875 0.001622 4.2e-05"/>
        <mass value="0.595"/>
        <inertia ixx="0.000402747" ixy="8.709e-06" ixz="2.97e-07" iyy="0.000691123" iyz="-5.45e-07" izz="0.000487919"/>
        </inertial>
    </link>
    <joint name="RL_hip_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 0.081 0"/>
        <parent link="RL_hip"/>
        <child link="RL_thigh_shoulder"/>
    </joint>
    <!-- this link is only for collision -->
    <link name="RL_thigh_shoulder">
        <collision>
        <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.032" radius="0.041"/>
        </geometry>
        </collision>
    </link>
    <joint name="RL_thigh_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 0.0838 0"/>
        <parent link="RL_hip"/>
        <child link="RL_thigh"/>
        <axis xyz="0 1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-1.0471975511965976" upper="4.1887902047863905" velocity="21"/>
    </joint>
    <link name="RL_thigh">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/thigh.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.0245 0.034"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="-0.003574 -0.019529 -0.030323"/>
        <mass value="0.888"/>
        <inertia ixx="0.005251806" ixy="-2.168e-06" ixz="0.000346889" iyy="0.005000475" iyz="-2.8174e-05" izz="0.0011102"/>
        </inertial>
    </link>
    <joint name="RL_calf_joint" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="RL_thigh"/>
        <child link="RL_calf"/>
        <axis xyz="0 1 0"/>
        <dynamics damping="0" friction="0"/>
        <limit effort="33.5" lower="-2.6965336943312392" upper="-0.9162978572970231" velocity="21"/>
    </joint>
    <link name="RL_calf">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://a1_description/meshes/calf.dae" scale="1 1 1"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.1"/>
        <geometry>
            <box size="0.2 0.016 0.016"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0.007105 -0.000239 -0.096933"/>
        <mass value="0.151"/>
        <inertia ixx="0.002344758" ixy="0.0" ixz="-0.000141275" iyy="0.002360755" iyz="0.0" izz="3.1158e-05"/>
        </inertial>
    </link>
    <joint name="RL_foot_fixed" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 -0.2"/>
        <parent link="RL_calf"/>
        <child link="RL_foot"/>
    </joint>
    <link name="RL_foot">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="orange"/>
        </visual>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <sphere radius="0.02"/>
        </geometry>
        </collision>
        <inertial>
        <mass value="0.06"/>
        <inertia ixx="9.600000000000001e-06" ixy="0.0" ixz="0.0" iyy="9.600000000000001e-06" iyz="0.0" izz="9.600000000000001e-06"/>
        </inertial>
    </link>
    <xacro:leg_transmission name="RL"/>

    <!-- <xacro:stairs stairs="15" xpos="0" ypos="2.0" zpos="0" /> -->

</robot>
